pkg/cloud/baremetal/actuators/machine/actuator_test.go

```diff
@@ -1630,3 +1632,203 @@ func TestDeleteOfBareMetalHostDeletesMachine(t *testing.T) {
 		}
 	}
 }
+
+func TestRemediation(t *testing.T) {
+	machine, machineNamespacedName := getMachine()
+	host, hostNamespacedName := getBareMetalHost()
+	node, nodeNamespacedName := geteNode()
+	linkMachineAndNode(machine, node)
+	host.Status.PoweredOn = true
+
+	//starting test with machine that needs remediation
+	machine.Annotations = make(map[string]string)
+	machine.Annotations[HostAnnotation] = host.Namespace + "/" + host.Name
+	machine.Annotations[externalRemediationAnnotation] = ""
+
+	scheme := runtime.NewScheme()
+	clusterapis.AddToScheme(scheme)
+	bmoapis.AddToScheme(scheme)
+	corev1.AddToScheme(scheme)
+
+	c := fakeclient.NewFakeClientWithScheme(scheme)
+
+	actuator, err := NewActuator(ActuatorParams{
+		Client: c,
+	})
+	if err != nil {
+		t.Error(err)
+	}
+
+	c.Create(context.TODO(), machine)
+	c.Create(context.TODO(), host)
+	c.Create(context.TODO(), node)
+
+	err = actuator.Update(context.TODO(), nil, machine)
+	if err != nil {
+		t.Errorf("unexpected error %v", err)
+	}
+
+	host = &bmh.BareMetalHost{}
+	c.Get(context.TODO(), hostNamespacedName, host)
+	if !hasRebootAnnotation(host) {
+		t.Log("Expected reboot annotation on the host but none found")
+		t.Fail()
+	}
+
+	host.Status.PoweredOn = false
+	c.Update(context.TODO(), host)
+
+	err = actuator.Update(context.TODO(), nil, machine)
+	if err != nil {
+		t.Errorf("unexpected error %v", err)
+	}
+
+	machine = &machinev1.Machine{}
+	c.Get(context.TODO(), machineNamespacedName, machine)
+
+	if _, exists := machine.Annotations[remediationInProgressAnnotation]; !exists {
+		t.Log("Expected remediation-in-progress annotation to exist on machine but none found")
+		t.Fail()
+	}
+
+	err = actuator.Update(context.TODO(), nil, machine)
+	if err != nil {
+		t.Errorf("unexpected error %v", err)
+	}
+
+	node = &corev1.Node{}
+	err = c.Get(context.TODO(), nodeNamespacedName, node)
+	if !errors.IsNotFound(err) {
+		t.Log("Expected node to be deleted")
+		t.Fail()
+	}
+
+	err = actuator.Update(context.TODO(), nil, machine)
+	if err != nil {
+		t.Errorf("unexpected error %v", err)
+	}
```

Comment:
Some comments that indicate what we expected `actuator.Update()` to do each time might be helpful

Reply:
Each `actuator.Update()` call is followed by a check and a text which describes what was expected to be done.
Let me know if you still think it's valuable to copy this to a comment